allennlp/modules/token_embedders/pretrained_transformer_embedder.py

```diff
@@ -122,6 +124,7 @@ def __init__(
         self._num_added_tokens = self._num_added_start_tokens + self._num_added_end_tokens
 
         if not train_parameters:
+            self.transformer_model.eval()
             for param in self.transformer_model.parameters():
                 param.requires_grad = False
```

Comment:
This might not be necessary anymore

Reply:
Got it. [This Stack Overflow post](https://stackoverflow.com/questions/60018578/what-does-model-eval-do-in-pytorch) says that it's still best practice to turn off gradients computation. Thoughts?

Reply:
Reading that, sounds like we may want to use `with torch.no_grad()` in the forward pass?

I'm also curious what happens after this class is initialized as part of a submodule for a `Model`, when `Model.train()` is called. Will that revert the `.eval()` call within this submodule?

Reply:
Might be worth having a test for that

Reply:
Nice catch! I fixed this issue, albeit in a hacky way (https://stackoverflow.com/questions/61980943/how-can-i-keep-a-pytorch-submodule-in-eval-mode, https://stackoverflow.com/questions/394770/override-a-method-at-instance-level). Let me know what you think. I also added a test case to ensure that PretrainedTrransformerEmbedder's transformer model remains in eval mode even when module that instantiates it is switched to training mode.

Reply:
One thing you could do is just rename the existing `.forward()` method to `._forward()`, and then implement a new `.foward()` like:

```python
def forward(self, ...):
    if self.train_parameters:
        return self._forward(...)
    else:
        with torch.no_grad():
            return self._foward(...)
```

Reply:
... that said, I just realized there are other places in our code-base that would expect `requires_grad` to be `False` when this module is initialized, if we're not training it. So yea, I think we do need to keep these lines.